ui/src/containers/SampleFlexView.jsx

```diff
@@ -1,156 +1,245 @@
   }
 
-  onClickCell(event, idx, disabled) {
-    if (!disabled) {
-      this.props.filter({ cellFilter: `${idx + 1}` });
+  function isPuckSelected(cellID, puckID) {
+    return (
+      Number(filterOptions.cellFilter) === cellID &&
+      Number(filterOptions.puckFilter) === puckID
+    );
+  }
+
+  // Call displayPuckCellContextMenu only after Redux state updates
+  useEffect(() => {
+    if (pendingEvent) {
+      displayPuckCellContextMenu(
+        pendingEvent.event,
+        pendingEvent.menuID,
+        pendingEvent.cellID,
+        pendingEvent.puckID,
+      );
+      setPendingEvent(null);
+    }
+  }, [filterOptions, pendingEvent, displayPuckCellContextMenu]);
+
+  useEffect(() => {
+    if (filterOptions.cellFilter === '') {
+      dispatch(filterAction({ cellFilter: `${1}`, puckFilter: '' }));
     }
+  }, [filterOptions.cellFilter, dispatch]);
+
+  function handleDisplayPuckCellContextMenu(event, menuID, cellID, puckID) {
+    event.preventDefault();
+    handleClickonCellPuck(event, cellID, puckID === null ? '' : puckID);
+    setPendingEvent({ event, menuID, cellID, puckID });
     event.stopPropagation();
   }
 
-  isCellSelected(cell) {
-    let isCellSelected = false;
-    if (Number(this.props.filterOptions.cellFilter) === cell) {
-      isCellSelected = true;
-    }
-    return isCellSelected;
+  function getCellSlicePath(idxCell, radius = 8, centerX = 10, centerY = 10) {
+    // Calculate start and end angles for each slice
+    const angleStart = (idxCell * 360) / numberOfCell;
+    const angleEnd = ((idxCell + 1) * 360) / numberOfCell;
+
+    // Convert angles to radians
+    const startX = centerX + radius * Math.cos((angleStart * Math.PI) / 180);
+    const startY = centerY + radius * Math.sin((angleStart * Math.PI) / 180);
+    const endX = centerX + radius * Math.cos((angleEnd * Math.PI) / 180);
+    const endY = centerY + radius * Math.sin((angleEnd * Math.PI) / 180);
+
+    // Create each Cell like a pie slice path (triangle shape)
+    return `M${centerX},${centerY} L${startX},${startY} A${radius},${radius} 0 0,1 ${endX},${endY} Z`;
   }
 
-  renderCircle(nbCell, idx, isCellSelected) {
-    const cellOrder = [8, 7, 6, 5, 4, 3, 2, 1];
-    const percentage = (100 / nbCell) * cellOrder[idx];
-    const circumference = 2 * Math.PI * 5; // 2*pi*rayon
-    const strokeDash = (percentage * circumference) / 100;
-
-    const angle = 2 * Math.PI * (strokeDash / circumference);
-
-    const strokeDashArray = `${strokeDash} ${circumference}`;
-    const rgbColorList = [
-      '#e8ebee',
-      '#cdced1',
-      '#e8ebee',
-      '#cdced1',
-      '#e8ebee',
-      '#cdced1',
-      '#e8ebee',
-      '#cdced1',
-    ];
-
-    const disableClasse =
-      this.props.cellSampleList(idx + 1)[0].length > 0
-        ? 'has-sample'
-        : 'empty-cell';
-
-    const disabled = this.props.cellSampleList(idx + 1)[0].length <= 0;
+  function getCirclesInSlice(idxCell, centerX = 10, centerY = 10) {
```

Comment:
Some Value were Hard coded here.
i could apply a formula but it becomes more complicated.